### PR TITLE
Merge tests to develop

### PR DESCRIPTION
**Pull request by karanjaE**
What does this PR do?
It merges tests for the API to develop branch

### CODE REVIEW

File: app/test/test_api.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        test_app.config["TESTING"] = True
+        return test_app
+
+    def setUp(self):
+        fakes = Factory.create()
+        username = fakes.user_name()
+        password = fakes.password()
+
+    def test_create_bucketlist_after_auth(self):
+        response = self.client.post("/bucketlists/", data={})
```
Comment (ghost) [inline app/test/test_api.py]: How are you creating a bucketlist without data?

File: app/test/test_api.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        self.assertEqual(response.status_code, 201)
```
Comment (ghost) [inline app/test/test_api.py]: Where is the authorization taking place?

File: app/test/test_api.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        response = self.client.get("/bucketlists/", data={})
+        self.assertEqual(response.status_code, 200)
+
+    def test_fail_to_get_bucket_lists_before_auth(self):
+        response = self.client.get("/bucketlists/", data={})
+        self.assertEqual(response.status_code, 401)
+        self.assertIn("You have to log in first", response.data)
+
+    def test_get_bucketlist_by_id(self):
+        response = self.client.get("/bucketlists/<id>", data={})
```
Comment (ghost) [inline app/test/test_api.py]: Shouldn't `<id>` be an actual value?

File: app/test/test_base.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+# #
+# #     # log in the created user
+# #     response = client.post("/auth/login",
+# #                            data={"username": username, "password": password})
+# #
+# #     token = json.loads(response.data).get("token")
+# #     return token
+#
+#
+# change questions on proctor.
```
Comment (ghost) [inline app/test/test_base.py]: Avoid pushing commented-out code

File: app/api/bucketlists/api.py [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+            abort(403, message="that name has already been taken.")
+        new_bucket = Bucket(bucket_name=bucket_name, created_by=user_id)
+        db.session.add(new_bucket)
+        db.session.commit()
+        return({"Message": "Bucketlist created."}, 201)
+
+    def get(self):
+        """Retrievs all bucketlists"""
+        auth = request.headers.get("access-token")
+        q = request.args.get("q", "")
```
Comment (njirap) [inline app/api/bucketlists/api.py:56]: maybe name this variable better? `query` maybe?

**Pull request merged**